src/coord/src/coord.rs

```diff
@@ -2854,6 +2884,31 @@ pub async fn serve(
     })
     .map_err(|s| CoordError::Unstructured(anyhow!("{}", s)))?;
 
+    let (metric_scraper_handle, metric_scraper_tx) = if let Some(LoggingConfig {
+        granularity,
+        retain_readings_for,
+        ..
+    }) = logging
+    {
+        let (tx, rx) = std::sync::mpsc::channel();
+        let mut scraper = Scraper::new(
+            granularity,
+            retain_readings_for,
+            ::prometheus::default_registry(),
+            rx,
+            internal_cmd_tx.clone(),
+        );
+        let executor = TokioHandle::current();
+        let scraper_thread_handle = thread::spawn(move || {
+            let _executor_guard = executor.enter();
+            scraper.run();
+        })
+        .join_on_drop();
```

Comment:
I haven't thought about this carefully but what if we just ... didn't join on drop? it seems like the `Scraper` thread needs this `tx` so that the `Coordinator` can tell it to shut down but the `Coordinator` needs to tell it to shut down because the coordinator blocks until the Scraper shuts down, but if we revoke that, then we don't need to do any of the extra stuff as a consequence. I think this is also a place where making the Scraper an async task would help (but that can wait till after this pr)

Reply:
Agreed that this is a bit more involved than it needs for the happy path - I tried to stay as true as I could to the timestamper design there; the thing that might make it worth it for now is tests, I think: We spawn a bunch of coordinators and having each of those leave threads sitting around that can panic might not be great (though I guess we can just drop the tx channel instead of sending a message? That might make this whole mechanism less involved).

For async tasks, I think it's gonna be difficult. The coord doesn't run an executor, so we'd have to figure out how to fit all that together. LMK what you think - happy either way.

Reply:
lets merge this in rn and then async-ify / figure out how to simplify the coordinator as a followup (but can you make a task so we don't forget?)